subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +511,590 @@ export function _handleReserveDataUpdated(
+  if (debtBalances[1].minus(accountDebtBalance.vTokenBalance).lt(BIGINT_ZERO)) {
+    const interestRateType = InterestRateType.VARIABLE;
+    const borrowerPosition = new PositionManager(
+      liquidateeAccount,
+      debtTokenMarket,
+      PositionSide.BORROWER,
+      interestRateType
+    );
+
+    const vBorrowerPositionID = borrowerPosition.subtractPosition(
+      event,
+      protocol,
+      debtBalances[1],
+      TransactionType.LIQUIDATE,
+      debtTokenMarket.inputTokenPriceUSD
+    );
+    if (!vBorrowerPositionID) {
+      log.error(
+        "[_handleLiquidate] positionID is null for market {} account {}",
+        [debtTokenMarket.id.toHexString(), liquidatee.toHexString()]
+      );
+    } else {
+      liquidatedPositions.push(vBorrowerPositionID!);
+    }
   }
 
-  liquidate.position = positionId;
-  liquidate.blockNumber = event.block.number;
-  liquidate.timestamp = event.block.timestamp;
-  liquidate.liquidator = liquidator.toHexString();
-  liquidate.liquidatee = borrower.toHexString();
-  liquidate.market = market.id;
-  liquidate.hash = event.transaction.hash.toHexString();
-  liquidate.nonce = event.transaction.nonce;
-  liquidate.logIndex = event.logIndex.toI32();
-  liquidate.asset = debtAsset.id;
-  liquidate.amount = amount;
-  liquidate.amountUSD = amount
-    .toBigDecimal()
-    .div(exponentToBigDecimal(inputToken!.decimals))
-    .times(market.inputTokenPriceUSD);
-  liquidate.profitUSD = liquidate.amountUSD.minus(
-    debtToCover
-      .toBigDecimal()
-      .div(exponentToBigDecimal(debtAsset.decimals))
-      .times(repayTokenMarket.inputTokenPriceUSD)
-  );
+  liquidate.positions = liquidatedPositions;
   liquidate.save();
 
-  protocol.cumulativeLiquidateUSD = protocol.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
-  );
-  protocol.save();
-  market.cumulativeLiquidateUSD = market.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
+  accountDebtBalance.sTokenBalance = debtBalances[0];
+  accountDebtBalance.vTokenBalance = debtBalances[1];
+  accountDebtBalance.save();
+}
+
+export function _handleFlashLoan(
+  asset: Address,
+  amount: BigInt,
+  account: Address,
+  procotolData: ProtocolData,
+  event: ethereum.Event,
+  premiumAmount: BigInt,
+  flashloanPremium: _FlashLoanPremium
+): void {
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning("[_handleFlashLoan] market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
+  const manager = new DataManager(
+    market.id,
+    market.inputToken,
+    event,
+    procotolData
   );
-  market.save();
+  const tokenManager = new TokenManager(asset, event);
+  const amountUSD = tokenManager.getAmountUSD(amount);
+  manager.createFlashloan(asset, account, amount, amountUSD);
+
+  let reserveFactor = market.reserveFactor;
+  if (!reserveFactor) {
+    reserveFactor = BIGDECIMAL_ZERO;
+  }
+  const protocolRevenueShare = reserveFactor.div(BIGDECIMAL_HUNDRED);
```

Comment:
isn't reserve factor already normalized to a decimal and not percentage?

Reply:
good catch. I believe it has been represented as percentage in current 2.0.1 implementation: https://github.com/messari/subgraphs/blob/8d0999ea1f259e90e93a41725d73c49bfd79220a/subgraphs/aave-forks/src/mapping.ts#L217-L219